File: 1355-minimum-deletions-to-make-array-beautiful/minimum-deletions-to-make-array-beautiful.cs
public class Solution {
public int MinDeletion(int[] nums)
    {
        int n = nums.Length;
        List<int> list = new List<int>(nums);

        int i = 0;
        while (i < list.Count)
        {
            if (i % 2 == 0)
            {
                int curr = list[i];
                i++;

                while (i < list.Count && list[i] == curr)
                {
                    list.RemoveAt(i);
                }
            }
            else
            {
                i++;
            }
        }

        if (list.Count % 2 == 1)
        {
            list.RemoveAt(list.Count - 1);
        }

        return n - list.Count;
    }
}

File: 1355-minimum-deletions-to-make-array-beautiful/minimum-deletions-to-make-array-beautiful.java
class Solution {
    public int minDeletion(int[] nums) {
        var n = nums.length;
        var list = new ArrayList<Integer>();

        for (var num : nums)
            list.add(num);

        for (var i = 0; i < list.size();) {
            if (i % 2 == 0) {
                var curr = list.get(i);
                i++;

                while (i < list.size() && curr.equals(list.get(i))) {
                    list.remove(i);
                }
            } else
                i++;
        }

        if (list.size() % 2 == 1)
            list.removeLast();

        return n - list.size();
    }
}